### ISSUE TITLE
Improper handling of properties with underscores in filter 

### ISSUE THREAD

**Issue by johnymontana** [neo4j-graphql/neo4j-graphql-js#263]
Copied from [Apollo Sprectrum post](https://spectrum.chat/apollo/apollo-client/regarding-filtering-on-simple-objects-broken~61fe454c-2f7a-4967-967d-2099595ca8e0):

My simple filters are not translated to a correct cypher syntax. For example :

```
  User(filter:{first_name_starts_with:"Reb"}) {
    first_name
    last_name
    email
    is_active
    id
  }
}
```

translates to:

```
MATCH (\user`:`User`) WHERE (`user`.first = $filter.first_name_starts_with) RETURN `user` { .first_name , .last_name , .email , .is_active , .id } AS `user` { "offset": 0, "first": -1, "filter": { "first_name_starts_with": "Reb" } }`
```

as you can see it's trying to match user.first (splitted from first_name) to match the filter instead of user.first_name.

If I try using last_name:

```
{
  User(filter: { last_name_starts_with: "R" } ) {
    first_name
    last_name
    email
    is_active
    id
    customers {
      name
    }
  }
}
```

I get query:

```
MATCH (`user`:`User`) WHERE (`user`.last = $filter.last_name_starts_with) RETURN `user` { .first_name , .last_name , .email , .is_active , .id ,customers: [(`user`)-[:`HAS_CUSTOMER`]->(`user_customers`:`Customer`) | user_customers { .name }] } AS `user`
{
  "offset": 0,
  "first": -1,
  "filter": {
    "last_name_starts_with": "R"
  }
}
```

Please help, filtering on 1st level relational fields works normally like

```
{
  User(filter: { customers: { name_starts_with: "R" } }) {
    first_name
    last_name
    email
    is_active
    id
    customers {
      name
    }
  }
}
```
Please help. thanks in advance

**Comment by thisisdev-patrick** [comment]
Originally this was an issue I posted on Spectum. A suggestion would be to break the execution of makeAugmentedSchema if fields have underscores in them. Not sure if this the way to go since the recommendations state otherwise: https://neo4j.com/docs/cypher-manual/current/syntax/naming/ 
`An exception to this rule is using underscore, as given by my_variable.`

**Comment by thisisdev-patrick** [comment]
PR ready
https://github.com/neo4j-graphql/neo4j-graphql-js/pull/264

**Comment by johnymontana** [comment]
Fixed in v2.6.3 release
